cliconfig/config.go

```diff
@@ -29,20 +31,15 @@ var (
 	configDir = os.Getenv("DOCKER_CONFIG")
 )
 
-func getDefaultConfigDir(confFile string) string {
-	confDir := filepath.Join(homedir.Get(), confFile)
-	// if the directory doesn't exist, maybe we called docker with sudo
-	if _, err := os.Stat(configDir); err != nil {
-		if os.IsNotExist(err) {
-			return filepath.Join(homedir.GetWithSudoUser(), confFile)
-		}
-	}
-	return confDir
-}
-
 func init() {
 	if configDir == "" {
-		configDir = getDefaultConfigDir(".docker")
+		configDir = filepath.Join(homedir.Get(), defaultConfigDir)
+		if _, err := os.Stat(configDir); err != nil {
+			// if the directory doesn't exist, maybe we called docker with sudo
```

Comment:
Is it normal to default to the home dir of the person who called sudo? If this is the first time we're running docker on this host I would think we'd want the config files to be placed in ~root/.docker not ~user/.docker so that the next time someone else runs "sudo docker" we'd grab the same config file.

Note that if ~root/.docker and ~user/.docker do not exist then we'll return to ~user/.docker - seems like ~root/.docker would be more appropriate given (in the end) we're running as root.  no?